### PR TITLE
Clarified uids behavior in register_all

### PR DESCRIPTION
**Pull request by kroenlein**
# Citrine Python PR

## Description 
Update documentation to address concerns in [SPT-119](https://citrine.atlassian.net/browse/SPT-119).

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/data_concepts.py [hunk 1]
```diff
@@ -503,18 +503,26 @@ def register_all(self,
...
+        them on to the server.  This is required as otherwise there is no way to determine how
+        objects are related to each other.  When the registered objects are returned from the
+        server, the input GEMD objects will be updated with whichever uids & _citr_auto:: tags are
+        on the returned objects.  This means GEMD objects that already exist on the server will
+        be updated with all their on-platform uids and tags.
 
-        Using this method should yield significant improvements to write speed over separate
-        calls to `register`.
+        This method has the same behavior as `register`, except that all no models will be
+        written if any one of them is invalid.  Using this method should yield significant
```
Comment (dkao-citrine) [inline src/citrine/resources/data_concepts.py:519]: Is this supposed to say "except that no models will be written if..."?

**Comment by dkao-citrine** [comment]
Thank you for the clarification! I had a one question about a potential typo.

File: src/citrine/resources/data_concepts.py (hunk 1 above)
Comment (kroenlein) [reply src/citrine/resources/data_concepts.py:519]: Good catch.

File: src/citrine/resources/data_concepts.py [hunk 2]
```diff
@@ -503,18 +503,26 @@ def register_all(self,
...
+        If the GEMD objects have no UIDs, Citrine IDs will be assigned to them prior to passing
+        them on to the server.  This is required as otherwise there is no way to determine how
+        objects are related to each other.  When the registered objects are returned from the
+        server, the input GEMD objects will be updated with whichever uids & _citr_auto:: tags are
+        on the returned objects.  This means GEMD objects that already exist on the server will
+        be updated with all their on-platform uids and tags.
 
-        Using this method should yield significant improvements to write speed over separate
-        calls to `register`.
+        This method has the same behavior as `register`, except that all no models will be
```
Comment (kroenlein) [inline src/citrine/resources/data_concepts.py:518]:
```suggestion
        This method has the same behavior as `register`, except that no models will be
```

**Review by dkao-citrine — APPROVED** [review]

**Pull request merged**